yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -958,6 +957,8 @@ object Client extends Logging {
     val sparkConf = new SparkConf
 
     val args = new ClientArguments(argStrings, sparkConf)
+    // to maintan backwards-compatibility
+    sparkConf.setIfMissing("spark.executor.instances", args.numExecutors.toString)
```

Comment:
Maybe I'm missing something, but this ensures `spark.executor.instances` is always set in cluster mode, which effectively means we can never use dynamic allocation in cluster mode?